tests/vm_test.sh

```diff
@@ -166,4 +166,56 @@ function test_vm_umount()
   }
 }
 
+function test_vm_parse_options()
+{
+  unset options_values
+  declare -gA options_values
+  local output
+  local option_output
+
+  # test individual options
+  option_output="$(vm_help)"
+  unset options_values
+  declare -gA options_values
+  output="$(parse_vm_options -h)"
+  assertEquals "($LINENO)" "$option_output" "$output"
+
+  function vm_mount()
+  {
+    echo 'mount'
+  }
+
+  option_output="$(vm_mount)"
+  unset options_values
+  declare -gA options_values
+  output="$(parse_vm_options --mount)"
+  assertEquals "($LINENO)" "$option_output" "$output"
+
+  function vm_umount()
+  {
+    echo 'umount'
```

Comment:
Don't use `echo`, always use `printf` instead. I also noticed there are a few `echo`s that were introduced to this file after #439, please replace them as well.